GG/GG/Font.h

```diff
@@ -233,6 +241,40 @@ class GG_API Font
         mutable X cached_width;
     };
 
+    /** \brief Used to encapsulate a text and tokens together to be rendered
+        using GG::Font. */
+    class GG_API TextElements
+    {
+    public:
+        TextElements(const Font& font);
+        ~TextElements();
+
+        /** Return the constructed text.*/
+        const std::string& Text();
+        /** Return the constructed TextElements.*/
+        const std::vector<boost::shared_ptr<TextElement> >& Elements();
```

Comment:
Does this class do the assembly, or does it provide an interface that other code uses to add elements to the vector of TextElement ?